src/cpp/transport/test_UDPv4Transport.cpp

```diff
@@ -155,10 +156,19 @@ bool test_UDPv4Transport::packet_should_drop(const octet* send_buffer, uint32_t
                 CDRMessage::readUInt32(&cdrMessage, &sequence_number.low);
                 cdrMessage.pos = old_pos;
 
-                if((!drop_participant_builtin_topic_data_ && writer_id == c_EntityId_SPDPWriter) ||
-                        (!drop_publication_builtin_topic_data_ && writer_id == c_EntityId_SEDPPubWriter) ||
-                        (!drop_subscription_builtin_topic_data_ && writer_id == c_EntityId_SEDPSubWriter))
-                    return false;
+                if(writer_id == c_EntityId_SPDPWriter ||
```

Comment:
Changing functionality here. I don't if this is what you wanted.

Reply:
It is the behavior I want. If always_drop is enabled, then the builtin topic data is dropped. Else, if there is set a percent of dropping, depending on `drop_participant_builtin_topic_data`, this kind of submessages will be taken on account or not.

Reply:
Done